yt_dlp/extractor/rai.py

```diff
@@ -507,7 +507,7 @@ class RaiPlaySoundLiveIE(RaiPlaySoundIE):
 
 
 class RaiPlaySoundPlaylistIE(InfoExtractor):
-    _VALID_URL = r'(?P<base>https?://(?:www\.)?raiplaysound\.it/(?:programmi|playlist|audiolibri)/(?P<id>[^/?#&]+))'
+    _VALID_URL = r'(?P<base>https?://(?:www\.)?raiplaysound\.it/(?:programmi|playlist|audiolibri)/(?P<id>[^/?#&]+))/*(?P<extra_id>[^?#&]+)?$'
```

Comment:
I don't understand the `/*`

Reply:
`/?` should be better.

It's to catch urls with or without the `<extra_id>` part at the end of the regex, but leave the `/` out of the regex group `<extra_id>` so that I could make a more clean `if extra_id:` instead of doing additional checks to see if `extra_id` was only `/` or if it was actually a valid `extra_id` like `puntate/prima-stagione-1995`

The regex should allow the input of these urls:
`https://www.raiplaysound.it/programmi/ilruggitodelconiglio`
`https://www.raiplaysound.it/programmi/ilruggitodelconiglio/` (not a valid url, but I'd like to allow it and just not use the final slash)
`https://www.raiplaysound.it/programmi/ilruggitodelconiglio/puntate/prima-stagione-1995`

Reply:
```suggestion
    _VALID_URL = r'(?P<base>https?://(?:www\.)?raiplaysound\.it/(?:programmi|playlist|audiolibri)/(?P<id>[^/?#&]+))(?:/(?P<extra_id>[^?#&]+))?'
```